### PR TITLE
front-end-test : all in one commit

### PR DESCRIPTION
**Pull request by YannLefj**
done all a once in minimum time possible (still almost 2 hours) and didn't reread it all much (might be ton's of console.log, who knows)
Couldn't pushed on the main repo, had to do a fork, that's why the pull request

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+			border-collapse: collapse;
+		}
+
+		thead tr{
+			border-top: dashed blue 2px;
+			border-bottom: dashed blue 2px;
+
+		}
+
+		tbody:before{
```
Comment (mlohezic) [inline index.html:17]: why ?

**Comment by barzul-labrador** [comment]
There are intentionnally missing columns in the json.

It results here in bad table generation

File: index.html [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    <script type="text/javascript" src="data.json"></script>
+    <script type="text/javascript">
+    function sortTheDataByKey() {
+        const key = this.innerHTML;
+        const sortedData = data.sort((a, b) => a[key].localeCompare(b[key]));
+        createTable(sortedData);
+    }
+
+    function changePage() {
+        console.log(this.value)
```
Comment (mlohezic) [inline index.html:63]: oups ;)

File: index.html (hunk 1 above)
Comment (YannLefj) [reply index.html:17]: to get a space between the thead and the tbody (not the best way I know, but margins were not working)

File: index.html (hunk 2 above)
Comment (YannLefj) [reply index.html:63]: yep..

**Pull request closed without merging**